scrapy/exporters.py

```diff
@@ -214,7 +232,16 @@ def _write_headers_and_set_fields_to_export(self, item):
                     # use fields declared in Item
                     self.fields_to_export = list(item.fields.keys())
             row = list(self._build_row(self.fields_to_export))
-            self.csv_writer.writerow(row)
+            self._writerow(row)
+
+    def _writerow(self, row):
+        self.csv_writer.writerow(row)
+        if six.PY2:
+            data = self.stream.getvalue()
```

Comment:
Any blockers with this PR? Really want it to get merged, so I dont need to
use my own fork
On Jun 10, 2016 16:42, "Paul Tremberth" <notifications@github.com> wrote:

> In scrapy/exporters.py
> <https://github.com/scrapy/scrapy/pull/2034#discussion_r66624813>:
>
> > @@ -214,7 +232,16 @@ def _write_headers_and_set_fields_to_export(self, item):
> >                      # use fields declared in Item
> >                      self.fields_to_export = list(item.fields.keys())
> >              row = list(self._build_row(self.fields_to_export))
> > -            self.csv_writer.writerow(row)
> > +            self._writerow(row)
> > +
> > +    def _writerow(self, row):
> > +        self.csv_writer.writerow(row)
> > +        if six.PY2:
> > +            data = self.stream.getvalue()
>
> why are these lines below needed?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/scrapy/scrapy/pull/2034/files/d4715f452800af1ba82026a24097badbf8ff5274#r66624813>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AC-3KIWseHK6XP5mwN9QqdjyVI380HMvks5qKXfjgaJpZM4Iu3Yz>
> .
>